src/main/java/com/afornalik/ox/view/UIConsole.java

```diff
@@ -0,0 +1,59 @@
+package com.afornalik.ox.view;
+
+import com.afornalik.ox.model.board.Board;
+import com.afornalik.ox.view.draw.UIDrawBoard;
+
+import java.util.Objects;
+import java.util.Scanner;
+
+public class UIConsole implements UIOperations{
+
+    private final UIDrawBoard uiDrawBoard;
+    private final Scanner scanner;
+
+    public UIConsole(Scanner scanner) {
+        this.scanner = scanner;
+        uiDrawBoard = null;
```

Comment:
Nulls are dangerous. Maybe one class that have UIDrawBoard and other that does not? Or maybe even separate drawing board from ui?

Reply:
I make a lot of changes i extract two interfaces in top of which i created two class UISimple witch provide basic operations and UIExtended where You need to specify this uiDrawBoard parameter. There is no longer assigning to null.